### PR TITLE
Parse Github comments to trigger the CI on specific machines

### LINKED ISSUE #81: The CI has to recognize machines to retry from the PR comments

### PR DESCRIPTION
**Pull request by teojgo**
Closes #81

### CODE REVIEW

**Comment by teojgo** [comment]
@jenkins-cscs retry all

**Comment by vkarak** [comment]
@jenkins-cscs retry dom

**Comment by teojgo** [comment]
@jenkins-cscs retry daint leone monch kesch

File: Jenkinsfile [hunk 1]
```diff
@@ -27,8 +14,47 @@ stage('Initialization') {
...
+
+            if (!machinesToRun) {
+                println 'No machines were found. Aborting...'
+                currentBuild.result = 'ABORTED'
+                return
+            }
+
             currentBuild.result = 'SUCCESS'
         } catch(err) {
+            println err.toString()
```
Comment (teojgo) [inline Jenkinsfile]: test comment on code

**Comment by teojgo** [comment]
test comment

**Comment by teojgo** [comment]
@jenkins-cscs retry all

File: Jenkinsfile [hunk 2]
```diff
@@ -1,34 +1,64 @@
...
 def machinesToRun = machinesList
-def shortCommit
+def uniqueID
 
 stage('Initialization') {
     node('master') {
         try {
             def scmVars = checkout scm
-            shortCommit = scmVars.GIT_COMMIT[0..6]
+            uniqueID = "${scmVars.GIT_COMMIT[0..6]}${env.BUILD_ID}"
```
Comment (vkarak) [inline Jenkinsfile]: I would put a hyphen between the commit hash and the build ID, so as not to be confused as a single number.

File: Jenkinsfile [hunk 3]
```diff
@@ -1,34 +1,64 @@
...
+                return
+            }
+
+            machinesRequested = []
+            for (i = 2; i < splittedComment.size(); i++) {
+                machinesRequested.add(splittedComment[i])
+            }
+
+            machinesToRun = machinesRequested.findAll({it in machinesList})
+
```
Comment (vkarak) [inline Jenkinsfile]: I would remove this line.

File: Jenkinsfile [hunk 4]
```diff
@@ -1,34 +1,64 @@
...
+            echo sh(script: 'env|sort', returnStdout: true)
+
+            if (githubComment == null) {
+                machinesToRun = machinesList
+                currentBuild.result == 'SUCCESS'
+                return
+            }
+
+            def splittedComment = githubComment.split()
+
```
Comment (vkarak) [inline Jenkinsfile]: I would remove this line.

**Pull request merged**